functions/notify_slack.py

```diff
@@ -119,10 +119,12 @@ def default_notification(subject, message):
         }
 
 def filter_message_from_slack(message):
-    if(message['source'] == "aws.iam" and type(message['detail']) is dict  and message['detail']['eventName'] == "GenerateCredentialReport"):
-        return True
+    if message.get('source', "") == "aws.iam" and message.get('detail', {}).get('eventName', '') == "GenerateCredentialReport":
+      return True
+    if message.get('source', "") == "aws.iam" and message.get('detail', {}).get('eventName', '') == "GenerateServiceLastAccessedDetails":
```

Comment:
This could probably just be
```python
    if message.get('source', "") == "aws.iam" and message.get('detail', {}).get('eventName', '') in ["GenerateCredentialReport", "GenerateServiceLastAccessedDetails"]:
```

Reply:
good idea!